syntax/scan.go

```diff
@@ -861,15 +864,26 @@ func (sc *scanner) scanNumber(val *tokenValue, c rune) Token {
 		return FLOAT
 	} else {
 		var err error
+		var ok bool = true
 		s := val.raw
 		if len(s) > 2 && s[0] == '0' && (s[1] == 'o' || s[1] == 'O') {
 			val.int, err = strconv.ParseInt(s[2:], 8, 64)
 		} else if len(s) > 2 && s[0] == '0' && (s[1] == 'b' || s[1] == 'B') {
 			val.int, err = strconv.ParseInt(s[2:], 2, 64)
 		} else {
+			// Checking int
 			val.int, err = strconv.ParseInt(s, 0, 64)
+			// Checking BigInt
+			if err != nil {
+				num := new(big.Int)
+				val.bigInt, ok = num.SetString(s, 0)
+				if ok {
```

Comment:
if ok { err = nil }

Otherwise, let the error from ParseInt fall through  to line 887.